src/dev/i18n/extract_handlebars_messages.test.js

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { extractHandlebarsMessages } from './extract_handlebars_messages';
+
+describe('dev/i18n/extract_handlebars_messages', () => {
+  it('extracts handlebars default messages', () => {
+    const handlebarsSourceBuffer = Buffer.from(`\
+window.onload = function () {
+  (function next() {
+    var failure = function () {
+      failure = function () {};
+
+      var err = document.createElement('h1');
+      err.style['color'] = 'white';
+      err.innerText = '{{i18n 'ui.id-1' \
+'{"defaultMessage": "Message text", "context": "Message context"}'}}';
+
+      document.body.innerHTML = err.outerHTML;
+    }
+  }());
+};
+`);
+
+    const actual = Array.from(extractHandlebarsMessages(handlebarsSourceBuffer));
+    const expected = [['ui.id-1', { message: 'Message text', context: 'Message context' }]];
+
+    expect(actual).toEqual(expected);
+  });
+
+  it('throws on wrong arguments amount', () => {
+    const handlebarsSourceBuffer = Buffer.from(`\
+window.onload = function () {
+  err.innerText = '{{i18n 'ui.id-1'}}';
+};
+`);
+
+    expect(() => extractHandlebarsMessages(handlebarsSourceBuffer).next()).toThrow(
+      'Wrong arguments amount for handlebars i18n call.'
+    );
+  });
+
+  it('throws on wrong properties argument type', () => {
+    const handlebarsSourceBuffer = Buffer.from(`\
+window.onload = function () {
+  err.innerText = '{{i18n 'ui.id-1' propertiesJSONIdentifier}}';
+};
+`);
+
+    expect(() => extractHandlebarsMessages(handlebarsSourceBuffer).next()).toThrow(
+      `Cannot parse "ui.id-1" message: properties string should be a string literal.`
+    );
+  });
+});
```

Comment:
__suggestion:__ can you please add a couple of tests for empty `id` and missing `defaultMessage` to make sure we throw in this case?